### DISCUSSION TITLE
How many rings in this molecule("C1(C2)CC(CC3)NC3CC2C1")?

### DISCUSSION THREAD

**Discussion by dalong12** [rdkit/rdkit#3793] [Q&A] [answered]
I think this mol has 3 rings, but rdkit shows me 4.  So I want to know which answer is right.
I use rdkit version 2020.09.3
The code is below
`from rdkit import Chem`
`from rdkit.Chem import Draw`
`from rdkit.Chem.Draw import rdMolDraw2D`
`from rdkit.Chem import rdDepictor`
`rdDepictor.SetPreferCoordGen(True)`
`from rdkit.Chem.Draw import IPythonConsole`
`from IPython.display import SVG`
`import rdkit`
`print(rdkit.__version__)`

2020.09.3
`smi = "C1(C2)CC(CC3)NC3CC2C1"`
`    mol = Chem.MolFromSmiles(smi)`
    `    ri = mol.GetRingInfo()`
    `    atominfo = ri.AtomRings()`
    `    print(atominfo)`
    `    d2d = rdMolDraw2D.MolDraw2DSVG(500,400)`
    `    d2d.drawOptions().addAtomIndices=True`
    `    d2d.DrawMolecule(mol)`
    `    d2d.FinishDrawing()`
    `    SVG(d2d.GetDrawingText())`
((1, 9, 10, 0), (4, 3, 6, 7, 5), (2, 3, 6, 7, 8, 9, 1, 0), (8, 9, 10, 0, 2, 3, 6, 7))

**Comment by greglandrum** [comment, marked as answer]
The RDKit applies a symmetrization procedure when there are multiple potential rings of the same size in the SSSR set.
There's a big of discussion of this in the documentation: http://www.rdkit.org/docs/RDKit_Book.html#ring-finding-and-sssr

In this case there are two 8-rings which could be included in the set of smallest rings:
![image](https://user-images.githubusercontent.com/540511/107231669-dc276880-6a20-11eb-9615-e8a4770dee9e.png)
and
![image](https://user-images.githubusercontent.com/540511/107231714-e77a9400-6a20-11eb-8121-ca33a7a61168.png)
Rather than randomly selecting one of them, the default ring finding code returns both of those.

**Reply by dalong12** [reply]
Thanks, Greg.
I need to count the spiro ring in each ring systems of one molecule.  If the GetRingInfo returns the ring atoms information without symm ringinfo, the result will seem fine to our medicinal chemists. Do you have any suggestion? 
Thanks very much. Greg.

**Reply by greglandrum** [reply]
I won't argue with you about what your chemists want to see, but I would be surprised if they thought that cubane has five rings and not six.
Still, if you want the normal SSSR results, you can do what the documentation I liked to above suggests and just call `GetSSSR()`

**Reply by greglandrum** [reply]
If, on the other hand, you just want the number of spiro atoms in a structure, you can use `rdMolDescriptors.CalcNumSpiroAtoms()`

**Reply by dalong12** [reply]
Thanks, Greg. Very helpful.
